### DISCUSSION TITLE
Does Win7 not support Transparent?

### DISCUSSION THREAD

**Discussion by wmchuang** [AvaloniaUI/Avalonia#11514] [Q&A] [answered]
![image](https://github.com/AvaloniaUI/Avalonia/assets/25163291/1b14b39d-d0af-4476-a9be-e783c00bbb0e)

**Comment by wmchuang** [comment]
https://github.com/AvaloniaUI/Avalonia/discussions/10683#discussioncomment-5337854  I saw this approach, so I also created a floating layer window,It runs normally on Win10, but is not transparent on Win7

**Reply by timunie** [reply, marked as answer]
I'm afraid that Windows 7 is not officially supported anymore. Neither by us, nor by Micrsoft. We also have no Win7 machine to test things. 

Note: Unlike other UI Frameworks Avalonia can still run on Win7, but newer Platformfeatures added in Win 10 are for sure not available on older ones. Also WPF has issues making a transparent Window.

**Reply by BinToss** [reply]
Would pull requests which add runtime conditions for Windows 7 compatibility be accepted? As shown in the other answer, some effects can work on Windows 7 via workarounds which utilize the older, but still supported window handle operations.

Actually, this could be implemented as a fallback solution for whenever the "modern" approach doesn't work as expected, regardless of the OSPlatform.

**Reply by maxkatz6** [reply]
Unlikely.

We had way too many hacks with transparency effects and window border parameters, that we are going to slowly remove most of them one by one.

In a case of transparency, the idea is to remove everything, except WinUI Composition (win10) and Direct Composition (win8+, yet to be implemented) APIs. But it's something yet to discuss internally after 11.0 release.

Win7 is not in our plans to support. The problem with WS_EX_LAYERED is that it is CPU-only, slow, incompatible with DirectX rendering. If these problems are not a big deal for your application, then please use these workarounds in your app.
For the context, https://stackoverflow.com/a/28770172/5007046 - why AllowTransparency is bad even in WPF (Direct/WinUI Composition solutions are way better in this matter).

**Comment by wmchuang** [comment]
ok I can understand . But I have many users using Win7, Does anyone have any ideas ? thanks

**Reply by maxkatz6** [reply]
You can get IntPtr handle from the window, like `window.PlatfromImpl.Handle` (in 0.10) or `window.TryGetPlatformHandle` (in 11.0).
With this handle you can call native win32 APIs for transparency.

**Reply by wmchuang** [reply]
```
var handle= _floatingContent.PlatformImpl.Handle.Handle;


 SetWindowLong(handle, (int)WindowLongParam.GWL_EXSTYLE, (uint)WindowStyles.WS_EX_LAYERED);


SetLayeredWindowAttributes(handle, ParseRGB(Brushes.Transparent.Color), 0,
            LayeredWindowFlags.LWA_COLORKEY);      
```

 thanks, I achieved it . Now, it is transparent on Win7
